### PR TITLE
ipc errors: include original callsite in stack

### PR DESCRIPTION
**Pull request by mribbons**
use local stack trace rather than the one generated by eval driven ipc result

### CODE REVIEW

**Comment by jwerle** [comment]
> use local stack trace rather than the one generated by eval driven ipc result

can you post the before and after of the stack trace for this change?

**Comment by mribbons** [comment]
```js
await fs.stat('gone.txt')
```

Before:
```js
Error: no such file or directory
    at request.onreadystatechange (file:///c:/app/build/win/App-dev-v1.0.0/socket/ipc.js:1330:31) {
  code: -4058,
  id: '17382172827888547173'
}
```

After:
```js
Error: no such file or directory
    at Error
    at Promise.reject (file:///c:/app/build/win/App-dev-v1.0.0/socket/util.js:152:23)
    at FileHandle.open (file:///c:/app/build/win/App-dev-v1.0.0/socket/fs/handle.js:399:29)
    at async FileHandle.open (file:///c:/app/build/win/App-dev-v1.0.0/socket/fs/handle.js:129:5)
    at async visit (file:///c:/app/build/win/App-dev-v1.0.0/socket/fs/promises.js:40:18)
    at async Module.stat (file:///c:/app/build/win/App-dev-v1.0.0/socket/fs/promises.js:285:10)
    at async file:///c:/app/build/win/App-dev-v1.0.0/index.js:45:1 {
  code: -4058,
  id: '17484166926684574242'
}
```

File: api/util.js [hunk 1]
```diff
@@ -149,6 +149,9 @@ export function InvertedPromise () {
       },
 
       reject (error) {
+        // ipc errors start stack trace from the event eval position
+        // overwriting the stack trace as below ensures that the stack trace traces back to the original call
+        error.stack = Error().stack
```
Comment (jwerle) [inline api/util.js:154]: Let's use [`Error.cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) for this!

```js
const err = new Error(error.message, { cause: error })
promise.error = err
reject(err)
return promise
```

Comment (mribbons) [reply api/util.js]: Much better!
```js
Error: no such file or directory
    at Promise.reject (file:///c:/app/build/win/App-dev-v1.0.0/socket/util.js:156:21)
    at FileHandle.open (file:///c:/app/build/win/App-dev-v1.0.0/socket/fs/handle.js:399:29)
    at async FileHandle.open (file:///c:/app/build/win/App-dev-v1.0.0/socket/fs/handle.js:129:5)
    at async visit (file:///c:/app/build/win/App-dev-v1.0.0/socket/fs/promises.js:40:18)
    at async Module.stat (file:///c:/app/build/win/App-dev-v1.0.0/socket/fs/promises.js:285:10)
    at async file:///c:/app/build/win/App-dev-v1.0.0/index.js:45:1 {
  [cause]:
    Error: no such file or directory
        at request.onreadystatechange (file:///c:/app/build/win/App-dev-v1.0.0/socket/ipc.js:1330:31) {      
      code: -4058,
      id: '1935240965371362504'
    }
}
```

**Review by jwerle — APPROVED** [review]

**Pull request merged**